package/base-files/files/etc/uci-defaults/30_reconfigure_wifi

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+[ -f "/tmp/wifi.reload" ] || exit 0
+
+logger "Reconfiguring the wifi to ensure it uses the /etc/board.json"
+rm -f /etc/config/wireless
+wifi config
+wifi reload
+rm "/tmp/wifi.reload"
```

Comment:
wifi is not up at this point. So, instead of deleting everything and running half of it again, just move your added commands from mac80211.sh here.

Reply:
Just move all _your_ code from mac80211.sh here. You can cycle over radios/interfaces with config_foreach, and simply identify 2.4 GHz band by htmode=g ...

Reply:
If we remove them from the mac80211.sh, then whenever the user/system does a wifi config, that commands would be missing, so at least should be on both places, right?
Afterall, uci-default is a single time execution while you are booting

Reply:
etc/uci-defaults will be executed during first boot before anything is started. Just think of it as an extended version of /bin/config_generate.
So, whatever the user does comes way later.
However, there are certain downstream projects that also use uci-defaults scripts to set up their stuff. Thus, your uci-defaults scripts should be early (starting with 10-20) while the downstream scripts should take care to be later.
Nevertheless, the whole uci-defaults execution should take place before anything is really started for the first time.

Reply:
Sorry, I think I got your last comment wrong:

> then whenever the user/system does a wifi config, that commands would be missing

Is wifi config supposed to delete /etc/config/wireless?

Reply:
This done now. What do you think?